src/typelib.erl

```diff
@@ -44,6 +46,7 @@ pp_type(Type = {type, _, bounded_fun, _}) ->
                           [{capture, all_but_first, list}, dotall]),
     "fun(" ++ S ++ ")";
 pp_type({var, _, TyVar}) ->
+    %% TODO: In type(), TyVar should be an atom but we use a string.
```

Comment:
hm, I guess this only works because `io:format("~s, [atom])` works. This should really be an `atom_to_list/1`.
good catch.

(or we can support both atom and string to avoid creating a lot of new atoms, but then it needs to be reflected both in type and code)

Reply:
Yes, you're right. What's a high realistic number of generated type variables for a function? If it's never more than say 5 in practice, perhaps we can generate var names in a way so that they can be reused, rather than creating new ones for each function. Then we can use atoms and avoid extending the type.... Otherwise, we better extend our type.

I found it using `make gradualize`, after fixing the previous type error in this function using type assertion. :-)